src/flowcean/transforms/match_sampling_rate.py

```diff
@@ -13,99 +13,166 @@ class MatchSamplingRate(Transform):
-            - ...
+        ...
+        environment.load()
+        data = environment.get_data()
+        transform = MatchSamplingRate(
+            reference_feature="feature_a",
+            feature_columns={
+                "feature_b": "linear",
+            },
+        )
+        transformed_data = transform.transform(data)
+        ...
     ```
 
     The resulting Dataframe after the transform is:
 
-    | time_feature_a | feature_a   | time_feature_b | feature_b    | constant |
-    | -------------- | ----------- | -------------- | ------------ | -------- |
-    | [0, 1, 2]      | [2, 1, 7]   | [0, 1, 2]       | [10, 15, 20]    | 1    |
-    | [0, 1, 2]      | [4, 1, 0]   | [0, 1, 2]       | [20, 30, 40]    | 2    |
+    | feature_a                          | feature_b                          | const |
+    | ---                                | ---                                | ---   |
+    | list[struct[datetime[us],struct[]] | list[struct[datetime[us],struct[]] | int   |
+    | -----------------------------------| -----------------------------------| ----- |
+    | [{2024-06-25 12:26:01.0,{1.2}},    | [{2024-06-25 12:26:00.0,{1.0}},    | 1     |
+    |  {2024-06-25 12:26:02.0,{2.4}},    |  {2024-06-25 12:26:01.0,{1.2}},    |       |
+    |  {2024-06-25 12:26:03.0,{3.6}},    |  {2024-06-25 12:26:02.0,{2.4}},    |       |
+    |  {2024-06-25 12:26:04.0,{4.8}}]    |  {2024-06-25 12:26:03.0,{3.6}}]    |       |
 
 
-
-    Note that the used feature `time_feature_b` is still
-    present in the DataFrame. To remove it use the `select` transform.
-    """
+    """  # noqa: E501
 
     def __init__(
         self,
-        reference_timestamps: str,
-        feature_columns_with_timestamps: dict[str, str],
+        reference_feature: str,
+        feature_columns: dict[str, str],
     ) -> None:
         """Initialize the MatchSamplingRate transform.
 
         Args:
-            reference_timestamps: Timestamps of the reference feature.
-            feature_columns_with_timestamps: Names of the features that are
-                getting interpolated with their respective original timestamp
-                feature names.
+            reference_feature: Reference timeseries feature.
+            feature_columns: key-value pairs of feature columns and
+                the interpolation method to use. The interpolation
+                method can be one of the following:
+                - 'linear'
+                - 'slerp' (to be implemented)
         """
-        self.reference_timestamps = reference_timestamps
-        self.feature_columns_with_timestamps = feature_columns_with_timestamps
+        self.reference_feature = reference_feature
+        self.feature_columns = feature_columns
 
     def transform(self, data: pl.DataFrame) -> pl.DataFrame:
+        """Transform the input DataFrame.
+
+        Args:
+            data: Input DataFrame.
+
+        Returns:
+            Transformed DataFrame.
+
+        """
```

Comment:
This is an override of a method of a parent class. Mark it as such and drop the docstring.
```suggestion
    @override
    def transform(self, data: pl.DataFrame) -> pl.DataFrame:
```

Reply:
accepted the suggestion